lightning_client.go

```diff
@@ -3098,3 +3177,88 @@ func (s *lightningClient) ListPermissions(
 
 	return result, nil
 }
+
+// unmarshallHop unmarshalls a single hop.
+func unmarshallHop(rpcHop *lnrpc.Hop) (*Hop, error) {
+	var pubKey *route.Vertex
+
+	if rpcHop.PubKey != "" {
+		vertex, err := route.NewVertexFromStr(rpcHop.PubKey)
+		if err != nil {
+			return nil, err
+		}
+		pubKey = &vertex
+	}
+
+	return &Hop{
+		ChannelID:        rpcHop.ChanId,
+		Capacity:         btcutil.Amount(rpcHop.ChanCapacity),
+		Expiry:           rpcHop.Expiry,
+		AmtToForwardMsat: lnwire.MilliSatoshi(rpcHop.AmtToForwardMsat),
+		FeeMsat:          lnwire.MilliSatoshi(rpcHop.FeeMsat),
+		PubKey:           pubKey,
+	}, nil
+}
+
+// QueryRoutes can query LND to return a route (with fees) between two vertices.
+func (s *lightningClient) QueryRoutes(ctx context.Context,
+	req QueryRoutesRequest) (*QueryRoutesResponse, error) {
+
+	rpcCtx, cancel := context.WithTimeout(ctx, s.timeout)
+	defer cancel()
+
+	rpcCtx = s.adminMac.WithMacaroonAuth(rpcCtx)
+	rpcReq := &lnrpc.QueryRoutesRequest{
+		PubKey:  req.PubKey.String(),
+		AmtMsat: int64(req.AmtMsat),
+		FeeLimit: &lnrpc.FeeLimit{
+			Limit: &lnrpc.FeeLimit_FixedMsat{
+				FixedMsat: int64(req.FeeLimitMsat),
+			},
+		},
+		UseMissionControl: req.UseMissionControl,
+	}
+
+	if req.Source != nil {
+		rpcReq.SourcePubKey = req.Source.String()
+	}
+
+	if req.MaxCltv != nil {
+		rpcReq.CltvLimit = *req.MaxCltv
+	}
+
+	if req.LastHop != nil {
+		rpcReq.LastHopPubkey = req.LastHop[:]
+	}
+
+	var err error
+	rpcReq.RouteHints, err = marshallRouteHints(req.RouteHints)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := s.client.QueryRoutes(rpcCtx, rpcReq)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(resp.Routes) == 0 {
+		return nil, fmt.Errorf("no route found")
```

Comment:
Can we define this error so that callers don't need to string match?

Reply:
"Repurposed" an error for this as it seems there were some unused one in the file.